Added bing_mobile_view module and util class for Issue  #137

vikas-kundu opened the pull request:
Gif of the module: https://asciinema.org/a/yhFY89KbumDBJVyanNr8n6T57

vikas-kundu committed:
Added bing_mobile_view module and util class

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,132 @@
+}
+
+def blacklist_check(self, image_data):
+	project_root = project_root = up(up(up(__file__)))
```
double `project_root`

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,132 @@
+	project_root = project_root = up(up(up(__file__)))
+	black_img_path = os.path.join(project_root, 'data', 'images', 'blacklist', f'{self.options["blacklist"]}.json')
+
+	if os.path.isfile(black_img_path) == False: #Check if blacklisted image exists?
```
better to have `not os.path.isfile()`

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,132 @@
+	blacklisted_image_data = json.load(open(black_img_path)) #Load the data of blacklisted image
+	
+	counter, run = 1, self.bing_mobile_view(self.options['url'])
+	while blacklisted_image_data["img-data"] == image_data and counter <= self.options['retries']:
```
`'img-data'`

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,132 @@
+	folder_name = splitted_url[0]
+	file_name = splitted_url[1]
+
+	project_root = project_root = up(up(up(__file__)))
```
maybe you can make `project_root` global

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,132 @@
+	project_root = project_root = up(up(up(__file__)))
+	folder_filepath = os.path.join(project_root, 'data', 'images', folder_name)
+
+	if os.path.isdir(folder_filepath) == False: #If no pre-existing folder for image, make one
```
`not os.path.isdir()`

vikas-kundu committed:
Minor changes to bing_mobile_view module

vikas-kundu commented:
@z3y50n Done.

z3y50n commented:
Give me some time to test this module and I will provide additional feedback or merge!

z3y50n commented on modules/osint/bing_mobile_view.py:
```diff
@@ -0,0 +1,134 @@
+		return image_data
+
+def url_split(url):
+	folder_name = re.findall(r"/{2}[\w\.-]+/{1}",url) #Find string between // and /
```
`re` is not defined

z3y50n commented:
I also got this error:
![image](https://user-images.githubusercontent.com/33282622/113680727-d8832b80-96c9-11eb-962b-6f77f79f081d.png)

vikas-kundu committed:
Fixed bugs

vikas-kundu commented:
@z3y50n This error was there due to url_split function. So, made the following changes:

1. Added re in module.
2. Added error detect if no image found.
3. Replace file and folder naming function with simple method.
4. Improved regex for image detection in util class.
5. Modified blacklist image format according to the new regex.

It should work now.

saeeddhqan committed:
minor changes

saeeddhqan committed:
minor changes

saeeddhqan committed:
minor changes

saeeddhqan commented on core/util/bing_mobile_view.py:
```diff
@@ -0,0 +1,56 @@
+		self.framework.verbose('[Bing Mobile View] Fetching mobile view of the URL...')
+		bing_api_url = 'https://www.bing.com/webmaster/tools/mobile-friendliness-result'
+		
+		self.rand_agent = True
```
I think you need to change the global options

vikas-kundu replied:
Done

vikas-kundu committed:
Changes to global options

z3y50n commented:
Getting this error now 
![image](https://user-images.githubusercontent.com/33282622/113752070-3db03e80-9715-11eb-9fd9-b2e595cc0ca0.png)

saeeddhqan committed:
Add bing mobile view module

saeeddhqan reviewed (commented):
@z3y50n After this, if there's no problem, merge it.
Thanks!

vikas-kundu commented:
> @z3y50n After this, if there's no problem, merge it.
> Thanks!

There is one more problem. Adding commit for it.

vikas-kundu committed:
Bug fix in run.screenshot function

vikas-kundu commented on modules/osint/bing_mobile_view.py:
```diff
@@ -71,7 +71,7 @@ def module_api(self):
 
 	run = self.bing_mobile_view(url)
 
-	if not run.screenshot(): # If some error has occured while taking screenshot, return output to avoid further errors.
```
Here, when screenshot is successful, it will return a None type. As a result, this evaluates to True and the module exits with empty output. So changed it back to ` if run.screenshot() == False `.

vikas-kundu commented:
You can check now, it will work fine.

z3y50n commented:
Everything seems ok now. I am merging. Thank you @vikas-kundu
